File: Leetcode 3074. Apple redistribution in boxes/Solution.java
import java.util.Arrays;

class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int totalApples = 0;
        int num_boxes=0;
        Arrays.sort(capacity);
        for(int a:apple) 
            totalApples += a;
        for(int i=capacity.length-1; i>=0; i--) {
            if(totalApples <= 0)
                break;
            totalApples -= capacity[i];
            num_boxes++;
        }
        return num_boxes;
    }
}

File: Leetcode 3074. Apple redistribution in boxes/Solution.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        num_boxes=0
        capacity.sort(reverse=True)
        total_apples = sum(apple)
        while total_apples > 0:
            total_apples -= capacity[num_boxes]
            num_boxes += 1
        return num_boxes
